src/test/java/fi/metatavu/metamind/test/functional/StoryTestsIT.java

```diff
@@ -1,18 +1,101 @@
 package fi.metatavu.metamind.test.functional;
 
-import org.junit.Test;
-
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
+import java.util.UUID;
+
+import org.junit.Test;
+
+import feign.FeignException;
+import fi.metatavu.metamind.client.model.Story;
 import fi.metatavu.metamind.test.functional.builder.TestBuilder;
 
 public class StoryTestsIT extends AbstractFunctionalTest {
-  
+
   @Test
   public void testCreateStory() throws Exception {
     try (TestBuilder builder = new TestBuilder()) {
       assertNotNull(builder.admin().stories().create("en", "test story"));
     }
   }
-  
+  // TODO Uncomment test when permissions are done
+//  @Test
+//  public void testCreateStoryPermissions() throws Exception {
+//    try (TestBuilder builder = new TestBuilder()) {
+//      builder.invalid().stories().assertCreateFailStatus(403, "en", "test story");
+//      builder.anonymous().stories().assertCreateFailStatus(401, "en", "test story");
+//    }
+//  }
+
+  @Test
+  public void testFindStory() throws Exception {
+    try (TestBuilder builder = new TestBuilder()) {
+      Story createdStory = builder.admin().stories().create("en", "test story");
+      builder.admin().stories().assertFindFailStatus(404, UUID.randomUUID());
+      Story foundStory = builder.admin().stories().findStory(createdStory.getId());
+      builder.admin().stories().assertStoriesEqual(createdStory, foundStory);
+    }
+  }
+
+  // TODO Uncomment test when permissions are done
+//  @Test
+//  public void testFindStoryPermissions() throws Exception {
+//    try (TestBuilder builder = new TestBuilder()) {
+//      Story createdStory = builder.admin().stories().create("en", "test story");
+//
+//      assertNotNull(builder.admin().stories().findStory(createdStory.getId()));
+//
+//      builder.invalid().stories().assertFindFailStatus(403, createdStory.getId());
+//      builder.anonymous().stories().assertFindFailStatus(401, createdStory.getId());
+//
+//    }
+//  }
+
+  @Test
+  public void testUpdateStory() throws Exception {
+    try (TestBuilder builder = new TestBuilder()) {
+      Story createdStory = builder.admin().stories().create("en", "test story");
+      builder.admin().stories().assertStoriesEqual(createdStory, builder.admin().stories().findStory(createdStory.getId()));
+
+      Story updateStory = new Story();
+      updateStory.setName(createdStory.getName());
+      updateStory = builder.admin().stories().findStory(createdStory.getId());
```

Comment:
updateStory is overridden with resource found from the server so the test does not actually test for updating